cloud/pom.xml

```diff
@@ -0,0 +1,158 @@
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.11</artifactId>
+    <version>2.1.0-SNAPSHOT</version>
+    <relativePath>../pom.xml</relativePath>
+  </parent>
+
+  <artifactId>spark-cloud_2.11</artifactId>
+  <packaging>jar</packaging>
+  <name>Spark Project Cloud Integration</name>
+  <description>
+    Contains support for cloud infrastructures, specifically the Hadoop JARs and
+    transitive dependencies needed to interact with the infrastructures.
+
+    Any project which explicitly depends upon the spark-cloud artifact will get the dependencies;
+    the exact versions of which will depend upon the hadoop version Spark was compiled against.
+
+    Hadoop 2.7:
+      hadoop-aws
+      aws-java-sdk-s3
+      hadoop-azure
+      azure-storage
+      hadoop-openstack
+
+    WARNING: the signatures of methods in aws-java-sdk/aws-java-sdk-s3 can change between versions:
+    use the same version against which Hadoop was compiled.
+
+  </description>
+  <properties>
+    <sbt.project.name>cloud</sbt.project.name>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <!--Used for test classes -->
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_${scala.binary.version}</artifactId>
+      <version>${project.version}</version>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+
+
+    <!-- Jets3t is needed for s3n and s3 classic to work-->
+    <dependency>
+      <groupId>net.java.dev.jets3t</groupId>
+      <artifactId>jets3t</artifactId>
+    </dependency>
+
+    <!-- Explicit listing of transitive deps that are shaded. Otherwise, odd compiler crashes. -->
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+    <!-- End of shaded deps. -->
+  </dependencies>
+
+  <build>
+    <outputDirectory>target/scala-${scala.binary.version}/classes</outputDirectory>
+    <testOutputDirectory>target/scala-${scala.binary.version}/test-classes</testOutputDirectory>
+  </build>
+
+  <profiles>
+
+    <!--
+      This profile is enabled automatically by the sbt build. It changes the scope for the guava
+      dependency, since we don't shade it in the artifacts generated by the sbt build.
+    -->
+    <profile>
+      <id>sbt</id>
+      <dependencies>
+        <dependency>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+          <scope>compile</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>hadoop-2.7</id>
+        <dependencies>
+          <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-aws</artifactId>
```

Comment:
1. s3n was in hadoop-common for 2.5 and earlier; moved into the new hadoop-aws JAR alongside s3 and s3a in 2.6. wasb:// only came in hadoop 2.7. If people really wanted 2.6.x + cloud, it'd get more complex as the cloud module would need 2.6 and 2.7 profiling internally. 

1. Regarding Licensing. There's a small licensing problem right now: [HADOOP-13794](https://issues.apache.org/jira/browse/HADOOP-13794). We can't release any new ASF artifacts with the versions of the amazon-aws SDK currently in Hadoop as it has a version of the org.json artifacts, which are now license-non-grata. For Hadoop 2.9 we're moving to newer AWS SDKs (more specifically, bumping up Jackson so the SDKs work), but don't want to force in a jackson update to 2.7.x. Either I build/release a patched AWS SDK JAR *or* a version of the latest AWS with its dependencies shaded. The jackson issue matters less in Spark as it has a version of jackson which works, but as every AWS SDK release tends to be API incompatible, switching is non trivial. This is why it'd be good to keep a `cloud` profile separate; avoid worrying about this while I sort out what to do about Hadoop 2.7.4